app/workers/audience_segment_refresh_worker.rb

```diff
@@ -0,0 +1,14 @@
+class AudienceSegmentRefreshWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :low_priority
+
+  def perform(*ids)
+    if ids.blank?
+      ids = DisplayAd.approved_and_published
+        .distinct(:audience_segment_id)
+        .pluck(:audience_segment_id).compact
+    end
+    AudienceSegment.where(id: ids).find_each(&:refresh!)
```

Comment:
I'd suggestion pass this responsibility into another worker, so the whole set of IDs does not get retried when some fails. That, and apply usage of `perform_bulk` for additional optimization.

Reply:
:+1: Refactored to this in [2d2342e](https://github.com/forem/forem/pull/19284/commits/2d2342e347a73cdd224407d4c3c9c9aefe2e7da5).